service/routes.py

```diff
@@ -80,6 +80,33 @@ def get_recommendations(product_id1, product_id2):
         raise NotFound("Recommendation for product id {} and {} was not found.".format(product_id1, product_id2))
     return make_response(jsonify(recommendation.serialize()), status.HTTP_200_OK)
 
+@app.route("/recommendations/products/<int:product_id1>/related-products/<int:product_id2>", methods=["PUT"])
+def update_recommendations(product_id1, product_id2):
+    """
+    update a relationship
+    This endpoint will update a relationship based the data in the body that is posted
+    """
+    app.logger.info("Request to update a ")
+    check_content_type("application/json")
+    recommendation = Recommendation()
+    recommendation.deserialize(request.get_json())
+    old_recommendation = recommendation.find(product_id1, product_id2)
+    if not old_recommendation:
+        raise BadRequest("No existing relationship between product {} and {}".format(product_id1, product_id2))
+    if recommendation.relationship == "GO_TOGETHER":
+        old_recommendation.relationship = Type.GO_TOGETHER
+    elif recommendation.relationship == "CROSS_SELL":
+        old_recommendation.relationship = Type.CROSS_SELL
+    elif recommendation.relationship == "UP_SELL":
+        old_recommendation.relationship = Type.UP_SELL
+    elif recommendation.relationship == "ACCESSORY":
+        old_recommendation.relationship = Type.ACCESSORY
```

Comment:
Just do old_recommendation.relationship = recommendation.relationship

Reply:
Yes old_recommendation.relationship = recommendation.relationship should be sufficient. Also here's also where you are missing coverage but I don't think you have to test every type UPDATE. Making it old_recommendation.relationship = recommendation.relationship should also help with testing coverage.